### ISSUE TITLE
Detect text in line level

### ISSUE THREAD

**Issue by cuongngm** [BADBADBADBOY/pytorchOCR#17]
which are model can help me detect text in line level (like ctpn) instead of word level?

**Comment by fxwfzsxyq** [comment]
> which are model can help me detect text in line level (like ctpn) instead of word level?

All algorithms can be used, depending on your annotation, but I suggest using DBnet, which has good speed and accuracy

**Comment by cuongngm** [comment]
tks, i will try

**Comment by cuongngm** [comment]
hi, I used the text recognition model for horizontal text case (crop the text box after detect by DBnet model as your recommend), i want to ask is DBNet model good for vertical text as below? and can i do recognition task with vertical text and skew text? 
Example:  
![ocr1](https://user-images.githubusercontent.com/53366035/112269299-c6a58f80-8caa-11eb-9438-052e654f5f98.jpg)

**Comment by fxwfzsxyq** [comment]
you can judge the text type (Horizontal and vertical) by height and width , and rotate the vertical text to Horizontal. and you can&nbsp;generate some similar data to train crnn



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年3月24日(星期三) 下午3:17
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [BADBADBADBOY/pytorchOCR] Detect text in line level (#17)





 
hi, I used the text recognition model for horizontal text case (crop the text box after detect by DBnet model as your recommend), i want to ask is DBNet model good for vertical text as below? and can i do recognition task with vertical text and skew text?
 Example:

**Comment by fxwfzsxyq** [comment]
for example










------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年3月24日(星期三) 下午3:17
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [BADBADBADBOY/pytorchOCR] Detect text in line level (#17)





 
hi, I used the text recognition model for horizontal text case (crop the text box after detect by DBnet model as your recommend), i want to ask is DBNet model good for vertical text as below? and can i do recognition task with vertical text and skew text?
 Example:

**Comment by cuongngm** [comment]
as far as i understand, when rotate the image from vertical to horizontal, the text also is rotated and may cause false recognition? Example, assume i crop the text box below, and rotate the image to horizontal for recognition, but word 'typing' will be rotated and false recognition? 
![maxresdefault](https://user-images.githubusercontent.com/53366035/112275013-b5ac4c80-8cb1-11eb-8992-72e50a9c8caa.jpg)

**Comment by fxwfzsxyq** [comment]
So you need to generate similar data, rotate it to the&nbsp;horizontal, and then add it to the training data of crnn




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年3月24日(星期三) 下午4:01
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [BADBADBADBOY/pytorchOCR] Detect text in line level (#17)






据我所知，当旋转图像从垂直到水平，文字也旋转，可能会造成错误的识别？例如，假设我裁剪下面的文本框，并将图像旋转到水平以进行识别，但是单词‘ typing ’将被旋转并错误识别？


—
您正在接收此消息，因为您已评论。
直接回复此邮件，在GitHub上查看，或取消订阅.

**Comment by cuongngm** [comment]
tks, and valid data (text) and generate data (text is rotated) are same label, aren't they?

**Comment by fxwfzsxyq** [comment]
Yes, they are the same label



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年3月24日(星期三) 下午4:16
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [BADBADBADBOY/pytorchOCR] Detect text in line level (#17)





 
tks, and valid data (text) and generate data (text is rotated) are same label, aren't they?
